qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -375,6 +375,12 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         """
         if self.method != "default" and self.method not in self.plugins.ext_plugins:
             raise TranspilerError("Specified method: %s not found in plugin list" % self.method)
+
+        # If there aren't any gates to synthesize in the circuit we can skip all the iteration
+        # and just return.
+        if not set(self._synth_gates).intersection(dag.count_ops()):
+            return dag
+
         # Return fast if we have no synth gates (ie user specified an empty
         # list or the synth gates are all in the basis
         if not self._synth_gates:
```

Comment:
I think the test immediately below the inserted one will now always fail and can probably be removed; it's covered by the new test (an empty set trivially has an empty intersection with any other set).

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/10788/commits/c0895a8f1a6c55b1f9382addc4ea0b71c3f76487